geth/node/status_node.go

```diff
@@ -87,53 +86,45 @@ func (n *StatusNode) Start(config *params.NodeConfig, services ...node.ServiceCo
 		return ErrNodeRunning
 	}
 
-	if err := n.createNode(config); err != nil {
-		return err
-	}
-	n.config = config
+	var err error
 
-	if err := n.start(services); err != nil {
+	statusDB, err := db.Create(config.DataDir, params.StatusDatabase)
+	if err != nil {
 		return err
 	}
 
-	if err := n.setupRPCClient(); err != nil {
+	n.db = statusDB
+
+	defer func() {
+		if err != nil {
+			n.db.Close() //nolint: errcheck
```

Comment:
It will panic if creating the database fails because `n.db` is `nil`.

Reply:
Do we have a unit test that checks what happens when the database fails to be created?

Reply:
nope, I think it's a good idea to add it